### PR TITLE
CRYP-32: Saved filters CRUD API

### PR DESCRIPTION
**Pull request by jason-gerard**
JIRA: https://cryptify.atlassian.net/browse/CRYP-32

## Changes
- Add Filter entity
- Add CRD API for filter along with error handling
- Add API parameter validation
- Logic for handling relative filter date ranges

### CODE REVIEW

**Review by alexbara2000 — APPROVED** [review]
Looks good

**Review by narroarrow — APPROVED** [review]
Looks good!

File: packages/api/src/services/filters.service.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    ) {}
+
+    async findAll(req: GetFiltersRequest): Promise<Filter[]> {
+        const filters = await this.filtersRepository.findBy({
+            userId: req.userId,
+            currencyType: req.currencyType,
+        });
+
+        function hydrateRange(filter: Filter): Filter {
+            if (filter.start === "curr") {
```
Comment (Andre-Ibrahim) [inline packages/api/src/services/filters.service.ts:24]: could maybe leave a comment there explaining the if statement in a future pr

File: packages/api/src/services/filters.service.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        const doesFilterExists = !!(await this.filtersRepository.findOneBy({
+            userId: req.userId,
+            name: req.name,
+            currencyType: req.currencyType,
+        }));
+        if (doesFilterExists) {
+            throw new BadRequestException(ERROR_FILTER_ALREADY_EXISTS);
+        }
+
+        const filter = new FilterBuilder()
```
Comment (Andre-Ibrahim) [inline packages/api/src/services/filters.service.ts:52]: I like the builder!

**Review by Andre-Ibrahim — COMMENTED** [review]
looks good!

**Pull request merged**